ParkingZoneApp.Tests/Controllers/Admin/ParkingZoneControllerUnitTests.cs

```diff
@@ -0,0 +1,311 @@
+                    .Returns(parkingZone);
+
+            //Act
+            var result = _controller.Details(parkingZone.Id);
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.IsType<ViewResult>(result);
+            Assert.Equal(JsonSerializer.Serialize(parkingZone), JsonSerializer.Serialize(((ViewResult)result).Model));
+            _parkingZoneServiceMock.Verify(x => x.GetById(parkingZone.Id), Times.Once);
+        }
+
+        [Fact]
+        public void GivenInvalidId_WhenGetDetailsIsCalled_ThenReturnNotFound()
+        {
+            //Arrange
+            _parkingZoneServiceMock.Setup(service => service.GetById(parkingZone.Id));
+
+            //Act
+            var result = _controller.Details(parkingZone.Id);
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.IsType<NotFoundResult>(result);
+            _parkingZoneServiceMock.Verify(x => x.GetById(parkingZone.Id), Times.Once);
+        }
+        #endregion
+
+        #region Create
+        [Fact]
+        public void GivenValidCreateVM_WhenPostCreateIsCalled_ThenModelStateIsTrueAndReturnsRedirectToIndex()
+        {
+            //Arrange
+            var createVM = new CreateVM()
+            {
+                Id = Guid.NewGuid(),
+                Name = "Test Name",
+                Address = "Test Address",
+                CreatedDate = new DateOnly(2024, 7, 7)
+            };
+
+            _parkingZoneServiceMock.Setup(x => x.Insert(parkingZone));
+
+            //Act
+            var result = _controller.Create(createVM);
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.Equal("Index", ((RedirectToActionResult)result).ActionName);
+            Assert.True(_controller.ModelState.IsValid);
+            _parkingZoneServiceMock.Verify(service => service.Insert(It.IsAny<ParkingZone>()), Times.Once);
+        }
+
+        [Theory]
+        [InlineData("Name", "Required")]
+        [InlineData("Address", "Required")]
+        public void GivenCreateVM_WhenPostCreateIsCalled_ThenModelStateIsFalseAndReturnsViewResult(string key, string errorMessage)
+        {
+            //Arrange
+            CreateVM createVM = new CreateVM()
+            {
+                Id = parkingZone.Id,
+                Name = parkingZone.Name,
+                Address = parkingZone.Address,
+                CreatedDate = parkingZone.CreatedDate
+            };
+
+            _controller.ModelState.AddModelError(key, errorMessage);
+    
+            //Act
+            var result = _controller.Create(createVM);
+
+            //Assert
+            var model = Assert.IsType<ViewResult>(result).Model;
+            Assert.IsAssignableFrom<CreateVM>(model);
+            Assert.False(_controller.ModelState.IsValid);
+            Assert.Equal(JsonSerializer.Serialize(createVM), JsonSerializer.Serialize(model));
+        }
+
+        [Fact]
+        public void GivenValidCreateVM_WhenGetCreateIsCalled_ReturnsViewResult()
```

Comment:
write name of the test accorfing to that action here you didnt give anything  
![image](https://github.com/dilmurod-rozikov/ParkingZone/assets/86553376/833b5802-b256-4374-9ac7-9e4d524e2b03)